pkg/migration/controllers/migration.go

```diff
@@ -1536,6 +1487,100 @@ func (m *MigrationController) applyResources(
 				"Resource migrated successfully")
 		}
 	}
+	// apply remaining objects
+	for _, o := range updatedObjects {
+		metadata, err := meta.Accessor(o)
+		if err != nil {
+			return err
+		}
+		objectType, err := meta.TypeAccessor(o)
+		if err != nil {
+			return err
+		}
+		resource := &metav1.APIResource{
+			Name:       ruleset.Pluralize(strings.ToLower(objectType.GetKind())),
+			Namespaced: len(metadata.GetNamespace()) > 0,
+		}
+		var dynamicClient dynamic.ResourceInterface
+		if resource.Namespaced {
+			dynamicClient = remoteInterface.Resource(
+				o.GetObjectKind().GroupVersionKind().GroupVersion().WithResource(resource.Name)).Namespace(metadata.GetNamespace())
+		} else {
+			dynamicClient = remoteAdminInterface.Resource(
+				o.GetObjectKind().GroupVersionKind().GroupVersion().WithResource(resource.Name))
+		}
+
+		unstructured, ok := o.(*unstructured.Unstructured)
+		if !ok {
+			return fmt.Errorf("unable to cast object to unstructured: %v", o)
+		}
+
+		// set migration annotations
+		migrAnnot := metadata.GetAnnotations()
+		if migrAnnot == nil {
+			migrAnnot = make(map[string]string)
+		}
+		migrAnnot[StorkMigrationAnnotation] = "true"
+		migrAnnot[StorkMigrationName] = migration.GetName()
+		migrAnnot[StorkMigrationTime] = time.Now().Format(nameTimeSuffixFormat)
+		unstructured.SetAnnotations(migrAnnot)
+		retries := 0
+		log.MigrationLog(migration).Infof("Applying %v %v", objectType.GetKind(), metadata.GetName())
+		for {
+			_, err = dynamicClient.Create(context.TODO(), unstructured, metav1.CreateOptions{})
+			if err != nil && (errors.IsAlreadyExists(err) || strings.Contains(err.Error(), portallocator.ErrAllocated.Error())) {
+				switch objectType.GetKind() {
+				case "ServiceAccount":
+					err = m.checkAndUpdateDefaultSA(migration, o)
+				default:
+					// Delete the resource if it already exists on the destination
+					// cluster and try creating again
+					deleteStart := metav1.Now()
+					err = dynamicClient.Delete(context.TODO(), metadata.GetName(), metav1.DeleteOptions{})
+					if err != nil {
```

Comment:
Shouldn't there be a IsNotFound check here so that it falls into the else condition ?